kartothek/io/dask/dataframe.py

```diff
@@ -405,3 +406,61 @@ def collect_dataset_metadata(
         ddf = dd.from_pandas(df, npartitions=1)
 
     return ddf
+
+
+def _unpack_hash(df, unpack_meta):
+    df2 = unpack_payload_pandas(df, unpack_meta)
+    return _hash_partition(df2)
+
+
+def _hash_partition(part):
+    return pd.util.hash_pandas_object(part, index=False).sum().astype("uint64")
```

Comment:
I suppose the purpose of `uint64` is to prevent overflows here
```suggestion
    return pd.util.hash_pandas_object(part, index=False).astype("uint64").sum()
```

Reply:
No this is indeed expected. Summing the uint will cause overflows but these are handled gracefully. The result is then an int64 which may be negative. to stick with the practice of having uints (as is the case for the original pandas hash) I cast them back (which works also for negative numbers)


```python
In [7]: pd.Series([-1]).astype("uint64")
Out[7]:
0    18446744073709551615
dtype: uint64
```

Reply:
Makes sense. Just wondering (not suggesting), could we _prevent_ overflows here?

Reply:
only if we'd use a hash function with a different domain. the pandas hash maps to the entire uint64 space, i.e. as soon as we apply a sum we're in overflow territory. Everything else would need it's dedicated implementation mapping to a smaller domain

Reply:
Even if we'd put it into `np.complex128` someway (for lack of `uint128`) we could still overflow when `len(part) >= 2^64 ~= 1.8^19`